packages/rum/src/viewCollection.ts

```diff
@@ -195,11 +195,60 @@ function trackTimings(lifeCycle: LifeCycle, callback: (timings: Timings) => void
   return { stop: stopPerformanceTracking }
 }
 
-function trackLoadingTime(lifeCycle: LifeCycle, callback: (loadingTimeValue: number | undefined) => void) {
-  const startTime: number = performance.now()
+function trackLoadingTime(
+  lifeCycle: LifeCycle,
+  loadingType: ViewLoadingType,
+  callback: (loadingTimeValue: number) => void
+) {
+  let expectedTiming = 1
+  const receivedTimings: number[] = []
+
+  let stopLoadEventLoadingTime = noop
+  if (loadingType === ViewLoadingType.INITIAL_LOAD) {
+    expectedTiming += 1
+    ;({ stop: stopLoadEventLoadingTime } = trackLoadEventLoadingTime(lifeCycle, onTimingValue))
+  }
+
+  const { stop: stopActivityLoadingTimeTracking } = trackActivityLoadingTime(lifeCycle, onTimingValue)
+
+  function onTimingValue(timingValue: number | undefined) {
+    expectedTiming -= 1
+    if (timingValue) {
+      receivedTimings.push(timingValue)
+    }
+
+    if (expectedTiming === 0 && receivedTimings.length) {
+      callback(Math.max(...receivedTimings))
+    }
+  }
+
+  return {
+    stop() {
+      stopActivityLoadingTimeTracking()
+      stopLoadEventLoadingTime()
+    },
+  }
+}
+
+function trackLoadEventLoadingTime(lifeCycle: LifeCycle, callback: (loadingTimeValue: number) => void) {
+  const { unsubscribe: stopPerformanceTracking } = lifeCycle.subscribe(
+    LifeCycleEventType.PERFORMANCE_ENTRY_COLLECTED,
+    (entry) => {
+      if (entry.entryType === 'navigation') {
+        const navigationEntry = entry as PerformanceNavigationTiming
+        callback(msToNs(navigationEntry.loadEventEnd))
+      }
+    }
+  )
+
+  return { stop: stopPerformanceTracking }
+}
+
+function trackActivityLoadingTime(lifeCycle: LifeCycle, callback: (loadingTimeValue: number | undefined) => void) {
+  const startTime = performance.now()
   const { stop: stopWaitIdlePageActivity } = waitIdlePageActivity(lifeCycle, (hadActivity, endTime) => {
     if (hadActivity) {
-      callback(endTime - startTime)
+      callback(msToNs(endTime - startTime))
```

Comment:
what about doing the conversion in rum.ts like for duration?
then we only have to do it once and don't need to add this extra noise in the test